src/Source/DoctrineORMQuerySourceIterator.php

```diff
@@ -43,7 +43,11 @@ public function current()
     {
         $current = $this->iterator->current();
 
-        return $this->getCurrentData($current[0]);
+        $data = $this->getCurrentData($current[0]);
+
+        $this->query->getEntityManager()->getUnitOfWork()->detach($current[0]);
```

Comment:
Not sure about this, but based on [batch processing](https://www.doctrine-project.org/projects/doctrine-orm/en/2.7/reference/batch-processing.html#iterating-results), maybe we could have an internal count and when is reached, we could call `EntityManager::clear()`?

Reply:
Indeed, `clear` seems to be the recommended way https://github.com/doctrine/orm/blob/2.7/UPGRADE.md#deprecated-entitymanagermerge-and-entitymanagerdetach-methods